library/components/TTooltip.vue

```diff
@@ -1,64 +1,122 @@
+			this.showTooltip = false;
+		});
 	},
 	methods: {
 		placeTooltip() {
+			// Calculate and place both tooltip and arrow.
+			const triggerElement = this.$refs.trigger;
+			const tooltipElement = this.$refs.tooltip;
+			const tooltipPositions = {};
+			const positions = { arrow: {}, tooltip: {} };
+
+			if (triggerElement && tooltipElement) {
+				const position = triggerElement.getBoundingClientRect();
 
-            // Place tooltip near parent according to `position` prop.
-			const parent = this.$parent.$el;
-            const tooltipPositions = {}
-			if (parent) {
-				const position = parent.getBoundingClientRect();
 				if (this.position === "top") {
-					// Push from bottom
-					tooltipPositions.top = "-" + (position.bottom + 15) + "px";
+					positions.tooltip.top =
+						position.top - (tooltipElement.clientHeight + 10) + "px";
+					positions.tooltip.left =
+						position.left - tooltipElement.clientWidth / 2 + "px";
+					positions.arrow.top = position.top - 20 + "px";
+					positions.arrow.left = position.left + "px";
 				} else if (this.position === "bottom") {
-					// Push from top
-					tooltipPositions.top = (parent.clientHeight + 15) + "px";
+					positions.tooltip.top =
+						position.top + triggerElement.clientHeight + 15 + "px";
+					positions.tooltip.left =
+						position.left - tooltipElement.clientWidth / 2 + "px";
+					positions.arrow.top =
+						position.top + triggerElement.clientHeight + 10 + "px";
+					positions.arrow.left = position.left + "px";
 				} else if (this.position === "left") {
-                    // Place on left side
-                    tooltipPositions.right = (position.width + 15) + "px";
-                } else {
-                    // Place on right side
-                    tooltipPositions.left = (position.width + 15) + "px";
-                }
+					positions.tooltip.top =
+						position.top - tooltipElement.clientHeight / 2 + "px";
+					positions.tooltip.left =
+						position.left - tooltipElement.clientWidth - 15 + "px";
+					positions.arrow.top = position.top + "px";
+					positions.arrow.left = position.left - 25 + "px";
+				} else {
+					positions.tooltip.top =
+						position.top - tooltipElement.clientHeight / 2 + "px";
+					positions.tooltip.left =
+						position.left + triggerElement.clientWidth + 15 + "px";
+					positions.arrow.top = position.top + "px";
+					positions.arrow.left =
+						position.left + triggerElement.clientWidth + 10 + "px";
+				}
 			}
-            this.tooltipPositions = tooltipPositions;
+			this.positions = positions;
+		},
+		handleMouseOver() {
+			if (this.isClickable) return;
+			this.show();
+		},
+		handleMouseLeave() {
+			if (this.isClickable) return;
+			this.hide();
+		},
+		handleClick() {
+			if (this.isClickable) {
+				this.show();
+			}
+		},
+		show() {
+			this.showTooltip = true;
+
+			// Glitchy behaviour without nextTick()
+			this.$nextTick(() => {
+				this.placeTooltip();
+			});
```

Comment:
Unfortunately unable to avoid this for now.

Reply:
Please explain further?